pkg/services/meta/blocks.go

```diff
@@ -35,46 +38,78 @@ func (m *Meta) handleBlock(b *block.Header) error {
 	m.stM.RLock()
 	defer m.stM.RUnlock()
 
+	evsByStorage := make(map[*containerStorage][]objEvent)
 	for _, n := range res.Application {
 		ev, err := parseObjNotification(n)
 		if err != nil {
 			l.Error("invalid object notification received", zap.Error(err))
 			continue
 		}
 
-		s, ok := m.storages[ev.cID]
+		if magic := uint32(ev.network.Uint64()); magic != m.magicNumber {
+			l.Warn("skipping object notification with wrong magic number", zap.Uint32("expected", m.magicNumber), zap.Uint32("got", magic))
+		}
+
+		st, ok := m.storages[ev.cID]
 		if !ok {
 			l.Debug("skipping object notification", zap.Stringer("inactual container", ev.cID))
 			continue
 		}
 
-		err = m.handleObjectNotification(s, ev)
-		if err != nil {
-			l.Error("handling object notification", zap.Error(err))
-			continue
+		evsByStorage[st] = append(evsByStorage[st], ev)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		for st, ee := range evsByStorage {
+			st.putMPTIndexes(ee)
+		}
+	}()
+	wg.Add(1)
+	go func() {
```

Comment:
Looks like we still can't make asynchronous processing correct here. I'll take it as is for now just to have something to test, but we'll get back to it.

Reply:
elaborate?

Reply:
We've talked about a _pipeline_ for this, it's pretty natural one, from initial processor to extended one. Currently a part of the processing is done beforehand and then two separate routines are launched. Technically a possibility. I doubt it's an optimal one. But we don't have any test results either to prove it either way, so at this stage it doesn't matter much.